src/Stratis.FederatedPeg.Features.FederationGateway/Interfaces/ICrossChainTransferStore.cs

```diff
@@ -21,6 +21,12 @@ public interface ICrossChainTransferStore : IDisposable
         /// </summary>
         void Start();
 
+        /// <summary>
+        /// For optimization purposes the current tip is not saved after receiving an empty block.
+        /// This needs to be called to ensure the tip is persisted.
+        /// </summary>
```

Comment:
Most block will not contain deposits I think, however we would want to be able to monitor realtime where a gateway node is. So I feel like every time a block is received that should be reflected in the state of the node, which I understand is stored in the CrossChainTransferStore :/
How much slower is it to persist the current tip at every block ?

Reply:
Considering it takes only a minute or two sync the entire chain (when no deposits) it should not be too long to wait for a status update. We could update the tip using a asynchronous loop (i.e. just call this method) but not sure its really required.